### PR TITLE
feat (api, cli): Pipeline scheduler

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: cli/cds/application/pipeline.go [hunk 1]
```diff
@@ -2,34 +2,69 @@ package application
...
-
-func init() {
-	applicationPipelineCmd.AddCommand(cmdApplicationShowPipeline())
-	applicationPipelineCmd.AddCommand(cmdApplicationAddPipeline())
-	applicationPipelineCmd.AddCommand(cmdApplicationRemovePipeline())
-}
+var (
+	applicationPipelineCmd = &cobra.Command{
+		Use:     "pipeline",
+		Short:   "",
```
Comment (yesnault) [inline cli/cds/application/pipeline.go]: fill or remove ?

File: cli/cds/application/pipeline_scheduler.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+func applicationPipelineSchedulerDelete(cmd *cobra.Command, args []string) {
+	if len(args) != 4 {
+		sdk.Exit("Wrong usage: %s\n", cmd.Short)
+	}
+
+	projectKey := args[0]
+	appName := args[1]
+	pipelineName := args[2]
+	IDs := args[3]
```
Comment (yesnault) [inline cli/cds/application/pipeline_scheduler.go]: ids ?

File: engine/api/test/test.go [hunk 3]
```diff
@@ -27,3 +60,143 @@ func Setup(testname string, t *testing.T) *sql.DB {
...
+	t.Logf("Insert Project %s", key)
+
+	g := sdk.Group{
+		Name: name + "-group",
+	}
+
+	eg, _ := group.LoadGroup(db, g.Name)
+	if eg != nil {
+		g = *eg
+	} else {
```
Comment (yesnault) [inline engine/api/test/test.go]: else if

File: engine/sql/migrations/006_pipeline_scheduler.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+ALTER TABLE pipeline_build ADD COLUMN scheduled_trigger BOOLEAN default FALSE;
+ALTER TABLE pipeline_history ADD COLUMN scheduled_trigger BOOLEAN default FALSE;
+
+-- +migrate Down
+
+DROP TABLE pipeline_scheduler_execution;
+DROP TABLE pipeline_scheduler;
+
+ALTER TABLE pipeline_build DROP COLUMN scheduled_trigger;
```
Comment (sguiheux) [inline engine/sql/migrations/006_pipeline_scheduler.sql]: missing drop column scheduled_trigger on pipeline history

Comment (fsamin) [inline engine/sql/migrations/006_pipeline_scheduler.sql]: fixed

File: engine/api/scheduler.go [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+	s.ApplicationID = app.ID
+	s.PipelineID = pip.ID
+	s.EnvironmentID = env.ID
+
+	if err := scheduler.Insert(database.DBMap(db), s); err != nil {
+		log.Warning("addSchedulerApplicationPipelineHandler> cannot insert scheduler : %s", err)
+		WriteError(w, r, err)
+		return
+	}
+
```
Comment (fsamin) [inline engine/api/scheduler.go]: will modify last update handler with username instead

File: engine/api/scheduler.go [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+	if env != nil {
+		sOld.EnvironmentID = env.ID
+	}
+
+	if err := scheduler.Update(database.DBMap(db), sOld); err != nil {
+		log.Warning("updateSchedulerApplicationPipelineHandler> %s", err)
+		WriteError(w, r, err)
+		return
+	}
+
```
Comment (fsamin) [inline engine/api/scheduler.go]: will modify last update handler with username instead

File: engine/api/scheduler.go [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+	//Delete all the things
+	if err := scheduler.Delete(database.DBMap(db), sOld); err != nil {
+		WriteError(w, r, err)
+		return
+	}
+}
```
Comment (fsamin) [inline engine/api/scheduler.go]: will modify last update handler with username instead

**Pull request merged**